code/modules/mob/living/simple_animal/abnormality/waw/parasite_tree.dm

```
@@ -0,0 +1,382 @@
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/death()
+	if(connected_abno)
+		connected_abno.minions -= src
+		connected_abno.endBreach()
+	for(var/atom/movable/AM in src)
+		AM.forceMove(get_turf(src))
+	..()
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/CanAttack(mob/living/carbon/human/the_target) //Your target has to be human and not have the tree curse.
+	if(isturf(the_target) || !the_target || the_target.type == /atom/movable/lighting_object) // bail out on invalids
+		return FALSE
+
+	if(ismob(the_target)) //Target is in godmode, ignore it.
+		var/mob/M = the_target
+		if(M.status_flags & GODMODE)
+			return FALSE
+
+	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
+		return FALSE
+
+	if(ishuman(the_target) && the_target.stat != DEAD && !the_target.has_status_effect(THE_TREE_CURSE))
+		return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/AttackingTarget(atom/attacked_target)
+	return OpenFire()
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/OpenFire()
+	if(ranged_cooldown > world.time)
+		return FALSE
+	ranged_cooldown = world.time + ranged_cooldown_time
+	playsound(get_turf(src), projectilesound, 10, 1)
+	var/smoke_test = locate(/obj/effect/particle_effect/smoke) in view(1, src)
+	if(!smoke_test)
+		var/datum/effect_system/smoke_spread/parasite_tree/S = new(get_turf(src))
+		S.set_up(5, get_turf(src))
+		S.start()
+		return TRUE
+	ranged_cooldown += 1 SECONDS
+
+//SMOKE EFFECT
+/obj/effect/particle_effect/smoke/parasite_tree
+	name = "thick noxious fumes"
+	color = "#AAFF00"
+	lifetime = 3
+	opaque = TRUE
+
+/obj/effect/particle_effect/smoke/parasite_tree/smoke_mob(mob/living/carbon/C)
+	if(!istype(C))
+		return FALSE
+	if(lifetime<1)
+		return FALSE
+	if(C.internal != null) //this part is edited so that rabbits with gas masks can be effected as well.
+		return FALSE
```

Im not understanding the comment here, does this mean using a gas mask actually helps against the particles? If so I definitely approve of that.

---

Oh yeah the default was that if you had a gas mask then you wouldnt be effected at all. I didnt know how prevelent gas masks were in the facility but ill test if simple masks also do this.